### ISSUE TITLE
A couple of questions about the paper

### ISSUE THREAD

**Issue by coldmanck** [Scarecrow0/BGNN-SGG#1]
Dear authors, 

First of all, thank you for the nice paper! The main idea of refining graph structure using estimated confidence is impressive. I'm looking forward to your source code release! :) 

I would like to have you clarify some questions:
- Where does the supervision of the confidence module (RCE), i.e., L_{rce} come from? You mentioned L_{rce} consists of L_m and L_b, what're the meanings of these two terms? 
- In Table 1, I feel it's unfair to boldlize your BGNN performance as, simply, your mRs are not superior to that of PCPL [52]; especially, they're using an inferior backbone, i.e., VGG16.
- Performance **without** graph constraint would be interesting to see & be compared with existing methods, as that is the fairer metric that relaxes the unreasonable constraint of only one relation is being predicted for each pair.
- Figure 4 is not mentioned in the paper. Further explanation and clarification are nice to be included in the main text.
- You mentioned that the hyperparameter ρ ("lo-") is -5, which is not within [0, 1] interval.

*Edit: I found further explanations regarding the supervision of RCE in the Appendix and it's clear now.

**Comment by Scarecrow0** [comment]
Thanks for interested in our work! The codebase will be released ASAP.

I will answer your question point by point.
- As you see in Appendix, the RCE module shares the same supervision with the relationship classifier.
- The bold performance is based on the comparison of SGG models with the same detector and backbone(X101-FPN + Faster RCNN) in our paper. I'm trying to reproduce the performance of PCPL on the same backbone for a more fair comparison. I may show the performance of PCPL on model_zoo in this repo.
- I agree that the non-graph constraint post-process is able to obtain a better quality SG since it can handle the co-occurrence of the VR. 
However, most of the previous works are mainly compared upon the metrics with graph constraints and those two metrics are proportioned according to our observation. To this end, we only show the graph constraints metric on paper. Our codebase has full evaluation metrics includes non-graph constraints.
- Thanks for indicating that miss reference of fig. 4, it is the illustration of our new proposed bi-level resampling.
- This is a typo that the ρ =1e-5, we found that the joint train of entities head and VR head can hurt the performance, so the entities features are almost suppressed.  Those two typos have been updated on arXiv version, thanks for your indication.

**Comment by coldmanck** [comment]
Thank you for your reply.
- For Table 1, it's okay for me that you mean that's compared with the same backbone; however, as the ResNeXt-101 is apparently superior to VGG16 & you're proposing a new model, you should implement a VGG16 version of BGNN for comparison. Otherwise, it doesn't make sense to purely list VGG16-based models' performance.
- Actually, ng-mR & mR is NOT proportional to each other: you might want to implement & check Tang's TDE [1] performance on these two metrics and you'd find some surprising results.

[1] Kaihua Tang, Yulei Niu, Jianqiang Huang, Jiaxin Shi, and Hanwang Zhang.2020. Unbiased scene graph generation from biased training. In Proceedings of the IEEE/CVF Conference on Computer Vision and Pattern Recognition. 3716–3725

**Comment by Scarecrow0** [comment]
Sounds interesting, the ng-mR is discussed rarely in previous works.
I will pay more attention to ng-mR & mR metric. Thanks for pointing this point.
